### PR TITLE
CORE-19009 Highest wrapping key generation

### PR DESCRIPTION
**Pull request by simon-johnson-r3**
Ensure that when signing keys are created the wrapping key with the highest generation count for the specified alias is always picked.

### CODE REVIEW

**Review by Sakpal — COMMENTED** [review]
LGTM

File: components/crypto/crypto-softhsm-impl/src/integrationTest/kotlin/repository/SigningRepositoryTest.kt [hunk 1]
```diff
@@ -103,15 +104,31 @@ class SigningRepositoryTest : CryptoRepositoryTest() {
...
-    private val wrappingKeys = mutableMapOf<EntityManagerFactory, WrappingKeyInfo>()
-    private fun saveWrappingKey(emf: EntityManagerFactory, alias: String): WrappingKeyInfo {
-        return wrappingKeys.computeIfAbsent(emf) {
+    data class UniqueAliasAndGeneration(
+        val emf: EntityManagerFactory,
+        val alias: String,
+        val generation: Int
+    )
+
+    private val uniqueWrappingKeys = mutableSetOf<UniqueAliasAndGeneration>()
```
Comment (anien) [inline components/crypto/crypto-softhsm-impl/src/integrationTest/kotlin/repository/SigningRepositoryTest.kt]: Is this used anywhere? We do add new alias+generation to it, but I can't see if we do anything with it further.

Comment (simon-johnson-r3) [reply components/crypto/crypto-softhsm-impl/src/integrationTest/kotlin/repository/SigningRepositoryTest.kt]: It's to make sure we never try and write the same wrapping key with the same alias and generation to the in memory db more than once. The framework does not clear the db between each test, but many tests call the same function with the same parameters to set up the wrapping key.

Comment (anien) [reply components/crypto/crypto-softhsm-impl/src/integrationTest/kotlin/repository/SigningRepositoryTest.kt]: Got it now. I can also see how it's used below. Saw the braces wrong!

File: components/crypto/crypto-softhsm-impl/src/integrationTest/kotlin/repository/SigningRepositoryTest.kt [hunk 2]
```diff
@@ -103,15 +104,31 @@ class SigningRepositoryTest : CryptoRepositoryTest() {
...
+
+    private fun saveWrappingKey(
+        emf: EntityManagerFactory,
+        alias: String,
+        generation: Int = 1
+    ): UUID {
+        val uuid = UUID.randomUUID()
```
Comment (anien) [inline components/crypto/crypto-softhsm-impl/src/integrationTest/kotlin/repository/SigningRepositoryTest.kt]: We don't seem to be returning the existing uuid if the wrapping key already exists in the `uniqueWrappingKeys`

**Review by anien — APPROVED** [review]
LGTM.

**Pull request merged**